diff --git a/python/pyproject.toml b/python/pyproject.toml
index 6b6b0aa..219753a 100755
--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -29,7 +29,7 @@ dependencies = [
   "blobfile==3.0.0",
   "build",
   "compressed-tensors==0.18.0",
-  "cuda-python>=13.0",
+  "cuda-python>=12,<13",
   # Transitive dep of flashinfer-python, pinned because 1.6.0rc6 shipped no cp310 linux x86_64 wheel.
   "cuda-tile==1.6.0rc5",
   "datasets",
@@ -39,7 +39,7 @@ dependencies = [
   "einops",
   "fastapi",
   "flash-attn-4>=4.0.0b18",
-  "flashinfer_python[cu13]==0.6.18", # keep it aligned with jit-cache version in Dockerfile
+  "flashinfer_python[cu12]==0.6.18", # keep it aligned with jit-cache version in Dockerfile
   "gguf",
   "humming-kernels[cu13]==0.1.12",
   "interegular",
@@ -52,7 +52,7 @@ dependencies = [
   "ninja",
   "numba==0.65.1",
   "numpy",
-  "nvidia-cutlass-dsl[cu13]==4.6.2",
+  "nvidia-cutlass-dsl==4.6.2",
   "nvidia-mathdx==25.6.0",
   "nvidia-ml-py",
   "openai==2.6.1",
